cloudify_agent/installer/__init__.py

```diff
@@ -329,6 +332,20 @@ def create_custom_env_file_on_target(self, environment):
     def move(self, source, target):
         shutil.move(source, target)
 
+    def _get_certificate_file(self):
+        cert_file = self.cloudify_agent['local_rest_cert_file']
+
+        if not os.path.exists(cert_file):
+            cert_content = self.cloudify_agent.get('rest_cert_content')
+
+            if not cert_content:
+                return None
+
+            with open(cert_file) as f:
+                f.write(cert_content.decode('string-escape'))
```

Comment:
the cert has "\\n" instead of newlines in it - ie. the two characters "backslash, lowercase letter N", instead of a single character "\n" - ie. "newline". `.decode('string-escape')` decodes the escape sequences, ie. turns "\\n" into "\n" (it would also eg. turn "\\r" into "\r", or "\\x20" into " " - but the cert is of course pem-encoded, so the only escape sequence inside it, is "\\n")